core/src/main/java/io/fabric8/maven/core/service/openshift/OpenshiftBuildService.java

```diff
@@ -257,17 +259,50 @@ private String createBuildConfig(KubernetesListBuilder builder, String buildName
                 .withNewMetadata()
                 .withName(buildName)
                 .endMetadata()
-                .withNewSpec()
-                .withNewSource()
-                .withType("Binary")
-                .endSource()
-                .withStrategy(buildStrategyResource)
-                .withOutput(buildOutput)
-                .endSpec()
+                .withSpec(getBuildConfigSpec(buildStrategyResource, buildOutput))
                 .endBuildConfigItem();
         return buildName;
     }
 
+    private BuildConfigSpec getBuildConfigSpec(BuildStrategy buildStrategyResource, BuildOutput buildOutput) {
+        BuildConfigSpecBuilder specBuilder = null;
+
+        // Check for BuildConfig resource fragment
+        File buildConfigResourceFragment = KubernetesResourceUtil.getResourceFragmentFromSource(config.getResourceDir(), config.getResourceConfig(), "buildconfig.yml", log);
+        if (buildConfigResourceFragment != null) {
+            BuildConfig buildConfigFragment = client.buildConfigs().load(buildConfigResourceFragment).get();
+            specBuilder = new BuildConfigSpecBuilder(buildConfigFragment.getSpec());
+        } else {
+            specBuilder = new BuildConfigSpecBuilder();
+        }
+
+        if (specBuilder.buildSource() == null) {
+            specBuilder.withNewSource()
+                    .withType("Binary")
```

Comment:
Can we have eliminate the hardcoding? Not just here, like in general. Maybe we can make an Enum and do a refactor wherever required.